flink-java/src/main/java/org/apache/flink/api/java/operators/GroupReduceOperator.java

```diff
@@ -193,7 +199,11 @@ public SingleInputSemanticProperties getSemanticProperties() {
 	protected GroupReduceOperatorBase<?, OUT, ?> translateToDataFlow(Operator<IN> input) {
 
 		String name = getName() != null ? getName() : "GroupReduce at " + defaultName;
-		
+
+		this.function = (function instanceof CombineFunction) ?
+			new CombineToGroupCombineWrapper((CombineFunction) function) :
```

Comment:
This does also add a bit of overhead to the `reduce` method because it is wrapped. However, this should be OK since it is called just once per group.